### PR TITLE
feat: use displayState (instead of state + shipment status) for order status calculations

### PR DESCRIPTION
**Pull request by mdole**
### Description

Exchange exposes a `displayState` field that obviates the need for the calculations currently being performed by Eigen's `getOrderStatus` 

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Dev changes

use displayState (instead of state + shipment status) for order status calculations - matt & adam

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by mdole** [comment]
OK just finished a pretty thorough review of all different displayStates except for in_transit (because that one requires a change from Arta to mark an order as picked up; if anyone knows of a way to fake that on our side I'm all ears) and things looked good. 

One change I did notice: a wire transfer w/ Arta shipping order is currently `Pending` on live but is `Payment processing` after our change. I would say that's a net good 🤷 

I will also cop to the fact that I did not test this on Android since I don't have a running sim rn. Spent some time on it and then decided it wasn't worth it since we literally didn't change layout. If you disagree holla at me.

**Comment by mdole** [comment]
@artsy/px-devs tagged you for review since it looks like this [originated with your team](https://github.com/artsy/eigen/pull/5480). Any 👀 appreciated!

File: src/app/Scenes/OrderHistory/OrderDetails/Components/TrackOrderSection.tsx [hunk 1]
```diff
@@ -31,18 +31,21 @@ export const TrackOrderSection: React.FC<Props> = ({ section }) => {
...
         </Text>
-        {!!shipment?.trackingNumber ? (
+        {!!shipment?.trackingNumber && (
           <Text testID="trackingNumber" variant="sm" color="black60">
             Tracking:&nbsp;
             <Text variant="sm" color="black60" weight="medium">
               {shipment?.trackingNumber}
             </Text>
           </Text>
-        ) : (
```
Comment (mdole) [inline src/app/Scenes/OrderHistory/OrderDetails/Components/TrackOrderSection.tsx:41]: This was a lil drive-by fix: currently we sometimes display "Tracking not available" any time we don't have _Arta_ tracking available, even if we have a "track package" button since the partner input tracking information.

File: src/app/Scenes/OrderHistory/OrderDetails/Components/OrderDetailsHeader.tsx [hunk 2]
```diff
@@ -76,15 +74,6 @@ export const OrderDetailsHeaderFragmentContainer = createFragmentContainer(Order
           __typename
         }
       }
-      lineItems(first: 1) {
```
Comment (jpotts244) [inline src/app/Scenes/OrderHistory/OrderDetails/Components/OrderDetailsHeader.tsx:79]: Nice!

File: src/app/utils/getOrderStatus.ts [hunk 3]
```diff
@@ -1,69 +1,28 @@
...
+    case "FULFILLED":
+    case "REFUNDED":
+    case "PROCESSING":
+    case "PROCESSING_APPROVAL":
+    case "IN_TRANSIT":
+    case "CANCELED":
+      return orderStatusesMap[displayState].toLowerCase()
+    default:
+      return "unknown"
   }
```
Comment (jpotts244) [inline src/app/utils/getOrderStatus.ts:27]: Nice improvement here!

**Review by jpotts244 — APPROVED** [review]
Thanks for the updates here, this LGTM!

In the case of `in_transit` states, you are correct in that we dont receive transit-y related updates from Arta in our staging environment. 

We can mimic the data on our side by updating the `shipment` itself in exchange database if you are interested in more testing. I would have offered to do it for you but I am not sure what object you are testing with! Feel free we can talk in the morning LMK!

TL;DR we can do something like this:
```ruby
s = Shipment.find('blargh')

s.update!(status: "in_transit", carrier_name: "UPS", tracking_number: "1zw475770393992448", tracking_url:
  "https://www.ups.com/track?loc=en_us&tracknum=1zw475770393992448&Requester=NS/trackdetails")
  
```

File: src/app/utils/getOrderStatus.ts [hunk 4]
```diff
@@ -1,69 +1,28 @@
...
+  switch (displayState) {
+    case "SUBMITTED":
+    case "APPROVED":
+    case "FULFILLED":
+    case "REFUNDED":
+    case "PROCESSING":
+    case "PROCESSING_APPROVAL":
+    case "IN_TRANSIT":
+    case "CANCELED":
+      return orderStatusesMap[displayState].toLowerCase()
```
Comment (iskounen) [inline src/app/utils/getOrderStatus.ts:24]: **nitpick:** We can save a little bit of paper by removing the `orderStatusesMap` and returning the lower-cased string for each case in this switch statement. It's a nitpick, but keeping that map makes this code 10% harder to read, without giving us anything in return.

**Review by iskounen — DISMISSED** [review]

**Comment by mdole** [comment]
Sweet! Thanks @jpotts244 , that worked like a charm. Things look great when the order is in transit.

For posterity, here's what I ran:

```rb
o = Order.find '2fa423a9-bab7-48dd-a2e1-227c1683088f'
s = Shipment.find_by(line_item_id: o.line_items.first.id)
s.update(status: 'in_transit', carrier_name: 'ups', tracking_number: "1zw475770393992448", tracking_url: "https://www.ups.com/track?loc=en_us&tracknum=1zw475770393992448&Requester=NS/trackdetails")
```

File: src/app/utils/getOrderStatus.ts (hunk 4 above)
Comment (mdole) [reply src/app/utils/getOrderStatus.ts]: done in https://github.com/artsy/eigen/pull/8275/commits/7c28b3f45f66c92c84286b57cb970a0256b19716
